### PR TITLE
Add URI system records for KDC

### PR DESCRIPTION
**Pull request by tiran**
MIT KRB5 1.15 introduced KDC service discovery with URI records.
_kerberos and _kpasswd URI records can provide TCP, UDP, and Kerberos
KDC-Proxy references. URI lookups take precedence over SRV lookups,
falling back to SRV lookups if no URI records are found.

See: https://web.mit.edu/kerberos/krb5-latest/doc/admin/realm_config.html#kdc-discovery
Fixes: https://pagure.io/freeipa/issue/8968
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipaserver/dns_data_management.py [hunk 1]
```diff
@@ -134,7 +151,35 @@ def __add_srv_records(
 
             rdataset = zone_obj.get_rdataset(
                 r_name, rdatatype.SRV, create=True)
-            rdataset.add(rd, ttl=86400)  # FIXME: use TTL from config
+            rdataset.add(rd, ttl=self.TTL)
```
[the remark this answers is not in the document]
Comment (tiran) [reply ipaserver/dns_data_management.py:154]: I have reduced TTL to 3600.

File: ipaserver/dns_data_management.py [hunk 2]
```diff
@@ -40,6 +41,20 @@
     (DNSName('_kpasswd._udp'), 464),
 )
 
+IPA_DEFAULT_MASTER_URI_REC = (
+    # URI record name, URI template
+
+    # MIT KRB5 1.15+ prefers URI records for service discovery
+    # scheme (always krb5srv)
+    # flags (empty or 'm' for master)
```
[the remark this answers is not in the document]
Comment (tiran) [reply ipaserver/dns_data_management.py]: +1

**Comment by rcritten** [comment]
LGTM

**Comment by rcritten** [comment]
Can you remove the temp commit? I'll re-add the ack then.

**Comment by rcritten** [comment]
master:

* f1c58fb6466fba516f3d3e50a1a982b0162ef902 Add URI system records for KDC

**Pull request closed without merging**